cmd/manila-csi-plugin/main.go

```diff
@@ -142,6 +158,10 @@ func main() {
 	cmd.PersistentFlags().StringVar(&nodeID, "nodeid", "", "this node's ID")
 	cmd.MarkPersistentFlagRequired("nodeid")
 
+	cmd.PersistentFlags().StringVar(&nodeAZ, "nodeaz", "", "this node's availability zone")
+
+	cmd.PersistentFlags().BoolVar(&withTopology, "with-topology", false, "cluster is topology-aware")
```

Comment:
Im wondering if this key is actually required? If no topology info are passed , the AccessiblityRequirements would be nil , so no harm done in that case? why is it needed to explicitly set this?

Reply:
The reasoning behind having an explicit flag for enabling topology awareness is:

Linking these documents for reference:
* https://kubernetes-csi.github.io/docs/topology.html#overview
* https://github.com/kubernetes-csi/external-provisioner#topology-support

Let's say there's only one Manila AZ `shared-storage-a` and it is accessible from all nodes of the cluster. Those are spread across Nova zones `nova-1`, `nova-2`, `nova-3`. If

Reply:
> If no topology info are passed , the AccessiblityRequirements would be nil

If [`allowedTopologies`](https://kubernetes.io/docs/concepts/storage/storage-classes/#allowed-topologies) are not defined in the StorageClass, the external-provisioner will actually aggregate all available topologies of the cluster and continue from that set - if that's what you meant? [external-provisioner: Topology support](https://github.com/kubernetes-csi/external-provisioner#topology-support) for reference.

My reasoning behind this flag was that by default, all compute nodes have accessibility to the Manila-backed storage - so no topology-resolution is needed. In those "rare" cases where this is not true, admins need to explicitly state this fact and choose `allowedTopologies` carefully.

@tombarron @gouthampacha would you agree to get rid of this flag? This would mean that topology awareness would be mandatory (i.e. `--with-topology` as a permanent option), provisioned volumes would be tagged with:
* if `allowedTopologies == nil`: all topology terms from the whole cluster (for the topology key `topology.csi.manila.openstack.org/zone`)
* if `allowedTopologies != nil`: topology terms from `allowedTopologies`

Looking back, it does seem a bit superfluous.

Reply:
--with-topology kind of shadows the options you'd set on the external provisioner itself (feature gates, Topology=true and `--strict-topology`), but, how do we declare the capability `VOLUME_ACCESSIBILITY_CONSTRAINTS` without something like this? If `nodeAZ` is set? Or will it always be declared?

Reply:
Yes, `VOLUME_ACCESSIBILITY_CONSTRAINTS` would be declared always.

`--nodeaz` is another thing though: previously I thought the value part of the topology term is allowed to be empty (so that we'd just do `--nodeaz=""` in the case where our cluster doesn't care about topology), now I'm not so sure:

An excerpt from [CreateVolume CSI spec](https://github.com/container-storage-interface/spec/blob/master/spec.md#createvolume) - down at the very bottom of the section:
```
// Each value (topological segment) MUST contain 1 or more strings.
// Each string MUST be 63 characters or less and begin and end with an
// alphanumeric character with '-', '_', '.', or alphanumerics in
// between.
message Topology {
  map<string, string> segments = 1;
}
```

Reply:
@gouthampacha Capability only tells that this driver supports topology. If user wants to use it or not, it depends on him. We dont need variable to advertise capability. Thats the feature we are providing.

Reply:
Ah, that makes sense. I was looking at the [code as it is written](https://github.com/kubernetes/cloud-provider-openstack/pull/939/files#diff-2220be4cd1a6283cbdae8088c8359a57R61-R69) and thinking we'd need to selectively advertise that capability. 

@gman0, ouch - does that mean that it has to always have a default/non-empty string? perhaps a default like "manila" will be harmless?